src/vs/workbench/services/positronConsole/browser/positronConsoleService.ts

```diff
@@ -2081,36 +2084,48 @@ class PositronConsoleInstance extends Disposable implements IPositronConsoleInst
 	/**
 	 * Executes code.
 	 * @param code The code to execute.
+	 * @param runtimeCodeExecutionMode Possible code execution modes for a language runtime
+	 * runtimeCodeExecutionMode
 	 */
-	private doExecuteCode(code: string) {
+	private doExecuteCode(code: string, runtimeCodeExecutionMode?: RuntimeCodeExecutionMode) {
 		// Create the ID for the code that will be executed.
 		const id = `fragment-${generateUuid()}`;
 
-		// Create the provisional ActivityItemInput.
-		const activityItemInput = new ActivityItemInput(
-			ActivityItemInputState.Provisional,
-			id,
-			id,
-			new Date(),
-			this._session.dynState.inputPrompt,
-			this._session.dynState.continuationPrompt,
-			code
-		);
+		// Fallback to an interactive exection mode if one is not provided
+		const codeExecutionMode = runtimeCodeExecutionMode || RuntimeCodeExecutionMode.Interactive;
```

Comment:
How would you feel about making the signature like this instead, so it's clearer from the top what the default is:

```typescript
private doExecuteCode(code: string, mode: RuntimeCodeExecutionMode = RuntimeCodeExecutionMode.Interactive)
```

Reply:
So much cleaner! Will do :)